jaas/src/main/java/com/griddynamics/jagger/jaas/config/ApplicationConfig.java

```diff
@@ -0,0 +1,22 @@
+package com.griddynamics.jagger.jaas.config;
+
+import com.griddynamics.jagger.dbapi.DatabaseService;
+import com.griddynamics.jagger.engine.e1.services.DataService;
+import com.griddynamics.jagger.engine.e1.services.DefaultDataService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.*;
+
+@PropertySources({@PropertySource("classpath:/basic/default.environment.properties")})
+@ImportResource({"classpath:/common/storage.rdb.client.conf.xml", "classpath:/spring/dbapi.config.xml"})
+@Configuration
+public class ApplicationConfig {
```

Comment:
I'm not sure, if it's stayed somewhere as a rule, but it's a good way to logically separate your configuration from application entry point. Class marked with @SpringBootApplication is an entry point of the application and its main role is bootstrapping. It can be used as configuration class also, but this seems to be breaking single responsibility principle.
It's the same reason why you're creating separate classes for rest controllers rather than putting them into the main class - it's also working, but you don't do this because you don't want to mess your code up. 

Here you can read the same - http://stackoverflow.com/questions/29146019/where-to-put-bean-in-spring-boot

And also if you look through spring boot documentation you'll find plenty of examples with AppConfig (http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/context/annotation/Configuration.html), but you'll hardly find any with configuration in main class, only if it's not hello-world example.

I'm just telling, that logical separation of any kind of code will never make it worse.